src/blocks/remote-data-container/utils/tracks.ts

```diff
@@ -0,0 +1,26 @@
+import { recordTracksEvent } from '@automattic/calypso-analytics';
+
+import { getTracksBaseProps } from '@/utils/localized-block-data';
+
+/**
+ * Send a tracks event with the given name and properties.
+ */
+export function sendTracksEvent( eventName: string, props: Record< string, unknown > ): void {
+	const vipBaseProps = window.VIP_TRACKS_BASE_PROPS;
```

Comment:
Is a global necessary? Why can't the Tracks library expose a way to register these base props?

Reply:
This global is used to pass information from PHP to the frontend. I did some search about a better way to pass this information but couldn't find anything.

Reply:
I understand, I just don't think mu-plugins should be responsible for this

Reply:
Agreed. The main reason of passing this information from MU Plugins is to have a consistent implementation across all of VIP.

> Can the Tracks library expose a way to register these base props?

By registering you mean?

Reply:
> The main reason of passing this information from MU Plugins is to have a consistent implementation across all of VIP.

I am only objecting to the JS global as the method of passing the base props. Currently we are injecting it into the `<head>` which is not ideal. Providing it as a global also provides little value, since each plugin still needs custom code (and types) to access the global and inject it. I would rather avoid the JS global and script in `<head>`, have the plugins load the base props via PHP, and then inject it into their own global (e.g., `REMOTE_DATA_BLOCKS_SETTINGS` for this plugin). Alternatively plugins could integrate via a REST endpoint or a custom library.

> By registering you mean?

Another approach would be to render the base props as meta tag(s) and have the client-side code read them (once) and include them on each event.

Reply:
> and then inject it into their own global (e.g., REMOTE_DATA_BLOCKS_SETTINGS for this plugin)

I will implement it in `REMOTE_DATA_BLOCKS_SETTINGS` const.

> Alternatively plugins could integrate via a REST endpoint or a custom library.

I have initially suggested it in P2 and Mo feedback on this is that using the REST API is more expensive then making the calls directly to Tracks from the client.

> Another approach would be to render the base props as meta tag(s) and have the client-side code read them (once) and include them on each event.

It's more parsing, which I think is better to avoid.

Reply:
> I will implement it in `REMOTE_DATA_BLOCKS` const.

Done. https://github.com/Automattic/remote-data-blocks/pull/185/commits/44a3a22af29ccd0e0eead8489809e8bfc9667723